sway-core/src/type_system/type_info.rs

```diff
@@ -609,6 +609,15 @@ impl TypeInfo {
         }
     }
 
+    pub fn can_safely_ignore(&self) -> bool {
+        match self {
+            TypeInfo::Tuple(fields) => fields.is_empty(),
```

Comment:
What about an `TypeInfo::Array` of `TypeInfo::Tuple`, could that be ignored? (does Sway even support that? 😅 )

Reply:
An empty tuple is actually Unit, which is why it's here with the others.  I don't think ignoring non-empty data structures is desirable?

Reply:
I've updated the method in [this commit](https://github.com/FuelLabs/sway/pull/2627/commits/8463fcc6c4e16a2e9795a4ef7c8d4710859dd197) to make the check more complete.